compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/PlatformInsets.skiko.kt

```diff
@@ -22,117 +22,44 @@ import androidx.compose.runtime.Stable
 import androidx.compose.ui.ExperimentalComposeUiApi
 import androidx.compose.ui.unit.Dp
 import androidx.compose.ui.unit.dp
-import androidx.compose.ui.unit.lerp
 
 /**
  * This class represents platform insets.
  */
-@ExperimentalComposeUiApi
-@Immutable
-class PlatformInsets(
-    @Stable
-    val left: Dp = 0.dp,
-    @Stable
-    val top: Dp = 0.dp,
-    @Stable
-    val right: Dp = 0.dp,
-    @Stable
-    val bottom: Dp = 0.dp,
-) {
-    companion object {
-        val Zero = PlatformInsets(0.dp, 0.dp, 0.dp, 0.dp)
-    }
-
-    override fun equals(other: Any?): Boolean {
-        if (this === other) return true
-        if (other !is PlatformInsets) return false
-
-        if (left != other.left) return false
-        if (top != other.top) return false
-        if (right != other.right) return false
-        if (bottom != other.bottom) return false
-
-        return true
-    }
-
-    override fun hashCode(): Int {
-        var result = left.hashCode()
-        result = 31 * result + top.hashCode()
-        result = 31 * result + right.hashCode()
-        result = 31 * result + bottom.hashCode()
-        return result
-    }
-
-    override fun toString(): String {
-        return "PlatformInsets(left=$left, top=$top, right=$right, bottom=$bottom)"
-    }
-}
-
-internal fun PlatformInsets.union(insets: PlatformInsets) = PlatformInsets(
-    left = maxOf(left, insets.left),
-    top = maxOf(top, insets.top),
-    right = maxOf(right, insets.right),
-    bottom = maxOf(bottom, insets.bottom)
-)
-
-internal fun PlatformInsets.exclude(insets: PlatformInsets) = PlatformInsets(
-    left = (left - insets.left).coerceAtLeast(0.dp),
-    top = (top - insets.top).coerceAtLeast(0.dp),
-    right = (right - insets.right).coerceAtLeast(0.dp),
-    bottom = (bottom - insets.bottom).coerceAtLeast(0.dp)
-)
-
-internal fun lerp(start: PlatformInsets, stop: PlatformInsets, fraction: Float) =
-    PlatformInsets(
-        left = lerp(start.left, stop.left, fraction),
-        right = lerp(start.right, stop.right, fraction),
-        top = lerp(start.top, stop.top, fraction),
-        bottom = lerp(start.bottom, stop.bottom, fraction)
-    )
+interface PlatformInsets {
+    /**
+     * The left inset in pixels.
```

Comment:
Why do we need pixels instead of `Dp` now?

Reply:
Here I contemplated about which to choose. But because in the layout / measuring api we get size of the content retrieved from platform also in pixels, WindowInsetsRulers expect the provided values to be in pixels and so do `WindowInsets` I think it would make most sense to use pixels here, too. But I don't really have a strong position on this, so if you think it should remain `Dp` I can update.

Reply:
It seems px fits good here